File: basic/2979/2979.cpp
#include <bits/stdc++.h>
using namespace std;

int a, b, c;
int start, finish;
int arr[100];

int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    cin >> a >> b >> c;

    for(int i=0; i<3; i++){
        cin >> start >> finish;
        for(int j=start; j<finish; j++){
            arr[j]++;
        }
    }

    int fee = 0;

    for(int i : arr){
        if(i==0) continue;
        else if(i==1) fee += a;
        else if(i==2) fee += 2*b;
        else fee+= 3*c;
    }

    cout << fee << "\n";
}

File: basic/2979/2979.py
import sys

a, b, c =  map(int, sys.stdin.readline().split())

arr = [0 for _ in range(100)]

for i in range(3):
    start, end = map(int, sys.stdin.readline().split())
    for j in range(start, end):
        arr[j] += 1

fee = 0
for cnt in arr:
    if cnt == 0:
        continue
    elif cnt == 1: 
        fee += a
    elif cnt == 2:
        fee += 2*b
    else:
        fee += 3*c

sys.stdout.write(str(fee))